WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/CouponList.kt

```diff
@@ -0,0 +1,222 @@
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Divider
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.woocommerce.android.R
+import com.woocommerce.android.extensions.capitalizeWords
+import com.woocommerce.android.ui.coupons.CouponListViewModel.CouponListState
+import com.woocommerce.android.ui.coupons.CouponListViewModel.CouponUi
+import com.woocommerce.android.util.StringUtils
+import java.lang.StringBuilder
+import java.math.BigDecimal
+import kotlin.random.Random
+
+@Composable
+fun CouponListContainer(viewModel: CouponListViewModel) {
+    val couponListState by viewModel.couponsState.observeAsState(CouponListState())
+
+    CouponListContainer(state = couponListState)
+}
+
+@Composable
+fun CouponListContainer(state: CouponListState) {
+    when {
+        state.coupons.isNotEmpty() -> CouponList(
+            coupons = state.coupons,
+            currencyCode = state.currencyCode
+        )
+    }
+}
+
+@Composable
+fun CouponList(coupons: List<CouponUi>, currencyCode: String?) {
+    LazyColumn(
+        verticalArrangement = Arrangement.spacedBy(0.dp),
+        modifier = Modifier
+            .background(color = colorResource(id = R.color.color_surface))
+    ) {
+        itemsIndexed(coupons) { index, coupon ->
+            CouponListItem(
+                coupon = coupon,
+                currencyCode = currencyCode,
+            )
+            if (index < coupons.lastIndex) {
+                Divider(
+                    modifier = Modifier
+                        .offset(x = 16.dp),
+                    color = colorResource(id = R.color.divider_color),
+                    thickness = 1.dp
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun CouponListItem(coupon: CouponUi, currencyCode: String?) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp, vertical = 8.dp),
+        verticalArrangement = Arrangement.spacedBy(0.dp)
+    ) {
+        coupon.code?.let {
+            Text(
+                text = it,
+                style = MaterialTheme.typography.h6,
+                color = colorResource(id = R.color.color_on_surface),
```

Comment:
Same as above
```suggestion
                color = MaterialTheme.colors.onSurface,
```

Just sharing a tip: Compose supports a way to auto-calculate content color, but it needs to use a parent that supports it, when using `Modifier.background`, we don't get this behavior, check: https://developer.android.com/jetpack/compose/themes/material#content-color

Reply:
Thanks for the tips!